lib/DeployLib.sol

```diff
@@ -12,133 +11,49 @@ import {Create2Factory} from "@hyperlane-xyz/core/contracts/Create2Factory.sol";
 import {TestRecipient} from "@hyperlane-xyz/core/contracts/test/TestRecipient.sol";
 
 library DeployLib {
-    function deploy(
-        ConfigLib.HyperlaneDomainConfig memory config,
-        ConfigLib.MultisigIsmConfig memory ismConfig
-    ) internal {
-        deployProxyAdmin(config);
-        deployIgp(config);
-        deployMailbox(config, ismConfig);
-        deployTestRecipient(config);
-    }
-
-    function deployProxyAdmin(
-        ConfigLib.HyperlaneDomainConfig memory config
-    ) private {
-        if (address(config.admin) == address(0)) {
-            config.admin = new ProxyAdmin();
-            console.log(
-                "ProxyAdmin deployed at address %s",
-                address(config.admin)
-            );
-            config.admin.transferOwnership(config.owner);
-        } else {
-            console.log(
-                "Found ProxyAdmin at address %s, skipping deployment",
-                address(config.admin)
-            );
-        }
+    // keys must be alphabetically sorted for JSON serialization
+    // see https://book.getfoundry.sh/cheatcodes/parse-json?highlight=serialize#decoding-json-objects-into-solidity-structs
+    struct Core {
+        ProxyAdmin admin;
+        Create2Factory create2;
+        InterchainGasPaymaster igp;
+        Mailbox mailbox;
+        TestRecipient testRecipient;
     }
 
-    function deployIgp(ConfigLib.HyperlaneDomainConfig memory config) private {
-        require(
-            address(config.admin) != address(0),
-            "Must deploy ProxyAdmin before InterchainGasPaymaster"
-        );
-        if (address(config.igp) == address(0)) {
-            InterchainGasPaymaster impl = new InterchainGasPaymaster();
-            bytes memory initData = abi.encodeCall(
-                InterchainGasPaymaster.initialize,
-                ()
-            );
-            TransparentUpgradeableProxy proxy = new TransparentUpgradeableProxy(
-                address(impl),
-                address(config.admin),
-                initData
-            );
-            console.log(
-                "InterchainGasPaymaster deployed at address %s",
-                address(proxy)
-            );
-            config.igp = InterchainGasPaymaster(address(proxy));
-            config.igp.transferOwnership(config.owner);
-        } else {
-            console.log(
-                "Found InterchainGasPaymaster at address %s, skipping deployment",
-                address(config.igp)
-            );
-        }
-    }
+    function deploy(ConfigLib.Core memory config) internal returns (Core memory) {
```

Comment:
I see you removed incremental deploys. Nbd as we don't really use them here, just curious to get your thoughts